core/events/events_block_move.ts

```diff
@@ -124,14 +114,17 @@ export class BlockMove extends BlockBase {
    */
   private currentLocation_(): BlockLocation {
     const workspace = this.getEventWorkspace_();
+    if (!this.blockId) {
+      throw new Error(
+          'The block ID is undefined. Either pass a block to ' +
+          'the constructor, or call fromJson');
+    }
     const block = workspace.getBlockById(this.blockId);
     const location = {} as BlockLocation;
     const parent = block!.getParent();
     if (parent) {
```

Comment:
Maybe just make this `if (parent && block)` to avoid the non-null assertion?

Reply:
Reorganized this a bit and removed the non-null assertions.